### ISSUE TITLE
The `ca` option is not passed to node-socket instances

### ISSUE THREAD

**Issue by captainmuppet** [emailjs/emailjs-tcp-socket#66]
CA certificates are not passed to the `net.tls.connect()` function and the socket can not be upgraded if the server uses a certificate that is not in the mozilla CA Certificate Store.

Use-case: IMAP connection using emailjs-imap-client to internal servers.

**Comment by captainmuppet** [comment]
Fixed by #67 

The webpack version installed by running `npm install` can not be used to build the project, but webpack@4.19.1 works fine.
Please consider adding a package-lock or npm-shrinkwrap file.

**Comment by felixhammerl** [comment]
The node socket does not consume this option... Add the missing cert to your cert store intead.
https://nodejs.org/dist/latest-v11.x/docs/api/net.html#net_socket_connect_options_connectlistener

**Comment by captainmuppet** [comment]
True, but `tls.connect` does. That's why I used it twice (`constructor` / `upgradeToSecure`).

Last entry for [tls.connect](https://nodejs.org/dist/latest-v11.x/docs/api/tls.html#tls_tls_connect_options_callback) options.
All properties used by [tls.createSecureContext()](https://nodejs.org/dist/latest-v11.x/docs/api/tls.html#tls_tls_createsecurecontext_options) can be passed directly to the connect method.

**Comment by felixhammerl** [comment]
Firefox CA store does not have any influence on your systen cert store because Firefox brings its own store. Use the keychain (on macOS) or whatever your respective system uses.

**Comment by captainmuppet** [comment]
node.js uses the Mozilla CA Certificate Store by default as mentioned on the tls page:

https://nodejs.org/dist/latest-v10.x/docs/api/tls.html
ca: Optionally override the trusted CA certificates. **Default is to trust the well-known CAs curated by Mozilla.**

While installing the certificate on the OS level would (could\*) solve the problem, it's also a security risk to accept all certificates from a CA I otherwise wouldn't trust. With the ca option, I only have trust the certificate where it's needed.

\* I'm not sure if node.js uses both the internal cert store and the OS provided certificates
